### PR TITLE
jdk8u302-b05 and XCode 12 support

### PR DESCRIPTION
**Pull request by benty-amzn**
Merge jdk8u302-b05 from upstream and apply patches needed to build with XCode 12.5 on macOS 11.3.1

### CODE REVIEW

File: installers/mac/tar/build.gradle [hunk 1]
```diff
@@ -57,6 +57,7 @@ task configureBuild(type: Exec) {
     // Platform specific flags
     def command = ['bash', 'configure',
             "--with-toolchain-type=clang",
+            "--with-zlib=system",
```
Comment (lutkerd) [inline installers/mac/tar/build.gradle]: Add this to the changelog.

**Review by lutkerd — APPROVED** [review]

File: installers/mac/tar/build.gradle (hunk 1 above)
Comment (ziyiluo) [reply installers/mac/tar/build.gradle]: Ben, just wondering if this line is essential? I checked `generated-configure.sh` at L51372. The default zlib pick is system:
```
  if test "x$OPENJDK_TARGET_OS" = xmacosx; then
    #
    # On macosx default is system...on others default is
    #
    DEFAULT_ZLIB=system
  fi
```

The reason I prefer not to keep it (if not necessary) is to keep the macOS configuration flags consistent between all Corretto versions.

Comment (benty-amzn) [reply installers/mac/tar/build.gradle]: I think it was leftover from previous testing. Removed.

**Review by cliveverghese — APPROVED** [review]

**Review by ziyiluo — APPROVED** [review]
Excellent work! 🎉

**Pull request merged**